### PR TITLE
Stability fixes

### PR DESCRIPTION
**Pull request by vikulin**

### CODE REVIEW

**Comment by mwarning** [comment]
wow, impressive.

A few questions:
- what features did you add?
- did your bugs did your fix?
- why are the commit messages starting with `1.`?
- what linter (how) did you run? I also like to apply that to the develoment branch.

File: app/build.gradle [hunk 1]
```diff
@@ -3,15 +3,16 @@ apply plugin: 'com.android.application'
...
     signingConfigs {
     }
-    compileSdkVersion 26
+    compileSdkVersion 31
     defaultConfig {
         applicationId "d.d.meshenger"
-        minSdkVersion 19
-        targetSdkVersion 26
-        versionCode 303
-        versionName "3.0.3"
```
Comment (mwarning) [inline app/build.gradle:12]: We can set it to version 3.1.0.

**Comment by vikulin** [comment]
> wow, impressive.
> 
> A few questions:
> 
>     * what features did you add?
Mostly this PR contains simplifications:
 - removed encryption since [RiV-mesh](https://github.com/RiV-chain/RiV-mesh) network supports it by default
 - removed MAC in interface list in Settings
 - upgraded and refactored WebRTC code
>     * did your bugs did your fix?
 - fixed unexpected interruption while calling
 >     * why are the commit messages starting with `1.`?
 - It's used for commit index
>     * what linter (how) did you run? I also like to apply that to the develoment branch.
 - Used AndroidStudio and tested on Android 5.0, Android 10 also showed good results

Generally this is preparation for re-brending and much more deep changes in GUI. If you prefer to keep the same name (meshenger) let discuss it in [telegram channel](https://t.me/rivchain).

**Comment by mwarning** [comment]
I see, but this cannot be merged here when removing encryption and mac support. It might be a good idea to make all this optional. But you can do whatever you want. :-)

**Comment by vikulin** [comment]
You might take several ideas and merge partially/manually. I will keep the
project as a fork respecting your afforts as a creator.

чт, 25 нояб. 2021 г. в 22:20, Moritz Warning ***@***.***>:

**Comment by mwarning** [comment]
I do not mind if you want to remove the fork relationship. I will see what bugfixes and features I can merge and say thank you.

File: app/src/main/java/d/d/meshenger/Utils.java [hunk 2]
```diff
@@ -289,8 +294,14 @@ public static boolean isIP(String address) {
                     addressList.add(new AddressEntry(addr.getHostAddress(), nif.getName(), addr.isMulticastAddress()));
                     if(fullAddress.indexOf('%')>0){
                         String subaddress = fullAddress.substring(0, fullAddress.indexOf('%'));
+                        if(isIPv6(subaddress)){
+                            subaddress="["+subaddress+"]";
```
Comment (mwarning) [inline app/src/main/java/d/d/meshenger/Utils.java]: Is this a problem somewhere?

**Comment by vikulin** [comment]
Yes, interface address may contains % symbol

чт, 25 нояб. 2021 г. в 23:36, Moritz Warning ***@***.***>:

**Comment by vikulin** [comment]
> I see, but this cannot be merged here when removing encryption and mac support. It might be a good idea to make all this optional. But you can do whatever you want. :-)

WebRTC from Google should support encryption. Isn't it?

**Comment by mwarning** [comment]
WebRTC itself is encrypted, but the initial key exchange is not.

**Comment by vikulin** [comment]
Are you referring to a MITM attack?

пт, 26 нояб. 2021 г. в 09:55, Moritz Warning ***@***.***>:

**Comment by mwarning** [comment]
Yes, for example MIM.

**Comment by opusforlife2** [comment]
@mwarning Will you be releasing a new version with code picked from here, then?

**Comment by mwarning** [comment]
@opusforlife2 I will try to integrate some things into the development branch which hopefully turn into a new release. But the problem is available time..

**Comment by vikulin** [comment]
> @opusforlife2 I will try to integrate some things into the development branch which hopefully turn into a new release. But the problem is available time..

Consider kotlin migration for all code which I have already done:

https://github.com/meshenger-app/meshenger-android/pull/55/files

File: app/build.gradle [hunk 3]
```diff
@@ -23,22 +28,27 @@ android {
...
+    testImplementation 'junit:junit:4.13.1'
+    androidTestImplementation 'androidx.test.ext:junit:1.1.3'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
     implementation('com.journeyapps:zxing-android-embedded:3.6.0') { transitive = false }
-    implementation 'com.google.zxing:core:3.2.0'
-    implementation 'org.webrtc:google-webrtc:1.0.24064'
-    implementation 'com.github.joshjdevl.libsodiumjni:libsodium-jni-aar:2.0.2'
-    implementation 'net.java.dev.jna:jna:4.5.2@aar'
-    implementation 'com.github.isabsent:filepicker:1.1.01'
+    //back to 3.3.0 instead of 3.4.0 sue to https://github.com/zxing/zxing/issues/1170
```
Comment (opusforlife2) [inline app/build.gradle:51]:
```suggestion
    //back to 3.3.0 instead of 3.4.0 due to https://github.com/zxing/zxing/issues/1170
```

**Comment by mwarning** [comment]
> > @opusforlife2 I will try to integrate some things into the development branch which hopefully turn into a new release. But the problem is available time..
> 
> Consider kotlin migration for all code which I have already done:
> 
> https://github.com/meshenger-app/meshenger-android/pull/55/files

I always wanted to migrate to Kotlin. :-)

**Comment by vikulin** [comment]
Done release [v0.4.0.7](https://github.com/RiV-chain/CupLink/releases/tag/v0.4.0.7):

1.Tested Kotlin version
2.Redesigned Choose Address form
3.Bufixes

**Comment by mwarning** [comment]
Hi @vikulin, I started to integrate some of your fixes into the repository.
Let's close this MR for now, since it won't be merged directly.

**Pull request closed without merging**